src/cljs/org/broadinstitute/firecloud_ui/common/components.cljs

```diff
@@ -359,3 +368,82 @@
       (if (blank? (entity "documentation"))
         [:div {:style {:fontStyle "italic" :fontSize "90%"}} "No documentation provided"]
         [:div {:style {:fontSize "90%"}} (entity "documentation")])])})
+
+(react/defc GCSFilePreviewLink
+  {:render
+   (fn [{:keys [props state refs this]}]
+     (assert (:bucket-name props) "No bucket name provided")
+     (assert (:object props) "No GCS object provided")
+     [:div {}
+      [:a {:href "javascript:;"
+           :onClick #(react/call :show-dialog this)}
+       (:gcs-uri props)]
+      (when (or (:show-dialog? @state) (:loading? @state))
+        (let [{:keys [data error]} (:response @state)
+              data-size (when data (data "size"))
+              labeled (fn [label content]
+                        [:div {}
+                         [:div {:style {:display "inline-block" :width 120}} (str label ": ")]
+                         [:span {} content]])]
+          [:div {:style {:position "fixed" :top 0 :left 0 :right 0 :bottom 0 :zIndex 9999
+                         :fontSize "initial" :fontWeight "initial"}}
+           [Dialog
```

Comment:
I put it in the OKCancelForm and it looks good, but now you can't right-click to rename the file.  I had to go back to the dummy anchor that the button programmatically clicks.